pkg/oc/cli/cmd/set/deployment_strategy.go

```diff
@@ -0,0 +1,483 @@
+	Image               string
+	Command             []string
+	Environment         []string
+
+	// StatefulSet
+	Partition int32
+}
+
+// NewCmdDeploymentStrategy implements the set deployment-strategy command
+func NewCmdDeploymentStrategy(fullName string, f *clientcmd.Factory, out, errOut io.Writer) *cobra.Command {
+	options := &DeploymentStrategyOptions{
+		Out: out,
+		Err: errOut,
+	}
+	cmd := &cobra.Command{
+		Use:     "deployment-strategy RESOURCE --recreate|--rolling|--on-delete|--custom",
+		Short:   "Update a deployment strategy",
+		Long:    deploymentStrategyLong,
+		Example: fmt.Sprintf(deploymentStrategyExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(options.Complete(f, cmd, args))
+			kcmdutil.CheckErr(options.Validate())
+			if err := options.Run(); err != nil {
+				// TODO: move me to kcmdutil
+				if err == kcmdutil.ErrExit {
+					os.Exit(1)
+				}
+				kcmdutil.CheckErr(err)
+			}
+		},
+	}
+
+	kcmdutil.AddPrinterFlags(cmd)
+
+	cmd.Flags().Int64Var(&options.TimeoutSeconds, "timeout-seconds", deployapi.DefaultRecreateTimeoutSeconds, "The time to wait for updates before giving up.")
+	cmd.Flags().Int64Var(&options.IntervalSeconds, "interval-seconds", deployapi.DefaultRollingIntervalSeconds, "The time to wait between polling deployment status after update in rolling strategy.")
+	cmd.Flags().Int64Var(&options.UpdatePeriodSeconds, "update-period-seconds", deployapi.DefaultRollingUpdatePeriodSeconds, "The time to wait between individual pod updates in rolling strategy.")
+	cmd.Flags().Int32Var(&options.Partition, "partition", 0, "A number that indicates the ordinal at which the StatefulSet should be partitioned.")
+	cmd.Flags().StringVar(&options.MaxUnavailable, "max-unavailable", "", "The maximum number of pods that can be unavailable during the update in rolling strategy.")
+	cmd.Flags().StringVar(&options.MaxSurge, "max-surge", "", "The maximum number of pods that can be scheduled above the original number of pods in rolling strategy.")
+	cmd.Flags().StringVar(&options.Image, "image", "", "An image which can carry out a custom deployment strategy.")
+	cmd.Flags().StringArrayVarP(&options.Environment, "environment", "e", options.Environment, "Environment variables to use in the custom deployment pod")
+
+	cmd.Flags().BoolVar(&options.Recreate, "recreate", options.Recreate, "Set a Recreate strategy")
+	cmd.Flags().BoolVar(&options.Rolling, "rolling", options.Rolling, "Set a Rolling strategy")
+	cmd.Flags().BoolVar(&options.Custom, "custom", options.Custom, "Set a Custom strategy")
+	cmd.Flags().BoolVar(&options.OnDelete, "on-delete", options.OnDelete, "Set a OnDelete strategy")
+
+	cmd.Flags().StringSliceVarP(&options.Filenames, "filename", "f", options.Filenames, "Filename, directory, or URL to file to use to edit the resource.")
+	cmd.Flags().BoolVar(&options.Local, "local", false, "If true, set deployment hook will NOT contact api-server but run locally.")
```

Comment:
I suspect the comment is from copy paste...

Reply:
yeah ;)